engine/runtime-integration-tests/src/test/scala/org/enso/interpreter/test/instrument/RuntimeServerTest.scala

```diff
@@ -844,6 +844,94 @@ class RuntimeServerTest
     )
   }
 
+  it should "send method pointer updates of autoscope constructors" in {
+    val contextId  = UUID.randomUUID()
+    val requestId  = UUID.randomUUID()
+    val moduleName = "Enso_Test.Test.Main"
+
+    val metadata = new Metadata
+    val idA      = metadata.addItem(52, 3, "aa")
+    val idB      = metadata.addItem(70, 6, "ab")
+    val idC      = metadata.addItem(88, 7, "ac")
+
+    val code =
+      """type T
+        |    A
+        |    B x
+        |    C y z
+        |
+        |main =
+        |    a = test ..A
+        |    b = test (..B 42)
+        |    test (..C a b)
+        |
+        |test t:T = t
+        |""".stripMargin.linesIterator.mkString("\n")
+    val contents = metadata.appendToCode(code)
+    val mainFile = context.writeMain(contents)
+
+    metadata.assertInCode(idA, code, "..A")
+    metadata.assertInCode(idB, code, "..B 42")
+    metadata.assertInCode(idC, code, "..C a b")
+
+    // create context
+    context.send(Api.Request(requestId, Api.CreateContextRequest(contextId)))
+    context.receive shouldEqual Some(
+      Api.Response(requestId, Api.CreateContextResponse(contextId))
+    )
+
+    // open file
+    context.send(
+      Api.Request(requestId, Api.OpenFileRequest(mainFile, contents))
+    )
+    context.receive shouldEqual Some(
+      Api.Response(Some(requestId), Api.OpenFileResponse)
+    )
+
+    // push main
+    context.send(
+      Api.Request(
+        requestId,
+        Api.PushContextRequest(
+          contextId,
+          Api.StackItem.ExplicitCall(
+            Api.MethodPointer(moduleName, moduleName, "main"),
+            None,
+            Vector()
+          )
+        )
+      )
+    )
+    context.receiveN(5) should contain theSameElementsAs Seq(
+      Api.Response(requestId, Api.PushContextResponse(contextId)),
+      TestMessages.update(
+        contextId,
+        idA,
+        "Enso_Test.Test.Main.T",
+        Api.MethodCall(
+          Api.MethodPointer("Enso_Test.Test.Main", "Enso_Test.Test.Main.T", "A")
```

Comment:
This test is a copy of the non-autoscoped constructors test. However I am not sure if we can expect exactly the same results. I am seeing:
```
List(Response(Some(fe656c99-c433-478e-bd3e-0ba44a16b68e), PushContextResponse(0e26726c-1f93-416e-b18a-1b5105e74319)), Response(None, ExpressionUpdates(0e26726c-1f93-416e-b18a-1b5105e74319, 

Set(ExpressionUpdate(00000000-0000-00aa-8bae-ed8bb3344d4a, Some("Standard.Builtins.Main.Function"), None, Vector(ExecutionTime(0)), false, true, Value(None, None))))), Response(None, 

ExpressionUpdates(0e26726c-1f93-416e-b18a-1b5105e74319, 

Set(ExpressionUpdate(00000000-0000-00ab-2a79-b706024b455d, Some("Standard.Builtins.Main.Function"), None, Vector(ExecutionTime(0)), false, true, Value(None, None))))), Response(None, 

ExpressionUpdates(0e26726c-1f93-416e-b18a-1b5105e74319, Set(

ExpressionUpdate(00000000-0000-00ac-a00c-7da891184fde, Some("Standard.Builtins.Main.Function"), None, Vector(ExecutionTime(0)), false, true, Value(None, None))))), Response(None, ExecutionComplete(0e26726c-1f93-416e-b18a-1b5105e74319)))
```
the `Some("Standard.Builtins.Main.Function")` object is the _autoscoped constructor_ - e.g. `UnresolvedConstructor` class - its type is `Function` as it will ultimately become a function when resolved.

However at this moment we _don't know the real constructor_ yet - it get's resolved only later - but the [later messages are ignored](https://github.com/enso-org/enso/pull/9452/files#r1531614574) currently. Not sure how to modify `ProgramExecutionSupport` to handle the _autoscoped constructors_ properly. Any ideas, @4e6?